packages/expo-permissions/ios/EXPermissions/EXPermissions.m

```diff
@@ -28,266 +29,256 @@ @implementation EXPermissions
-    UM_ENSURE_STRONGIFY(self);
-    
-    // save results for permission
-    BOOL isGranted = [EXPermissions statusForPermissions:permission] == EXPermissionStatusGranted;
-    permissions[permissionType] = [NSMutableDictionary dictionaryWithDictionary:permission];
-    permissions[permissionType][@"granted"] = @(isGranted);
-    
-    askForNextPermission();
-  };
-  
-  // ask for first permission
-  askForNextPermission();
+  return [self getPermissionUsingRequester:[self getPermissionRequesterForClass:requesterClass]];
 }
 
 - (NSDictionary *)getPermissionsForResource:(NSString *)type
 {
-  Class requesterClass = _requesters[type];
-  if (requesterClass) {
-    if ([requesterClass respondsToSelector:@selector(permissionsWithModuleRegistry:)]) {
-      return [requesterClass permissionsWithModuleRegistry:_moduleRegistry];
-    }
+  return [self getPermissionUsingRequester:_requesters[type]];
+}
 
-    return [requesterClass permissions];
+- (NSDictionary *)getPermissionUsingRequester:(id<UMPermissionsRequester>)requester
+{
+  if (requester) {
+    return [EXPermissions parsePermissionFromRequester:[requester getPermissions]];
   }
-
   return nil;
 }
 
 // shorthand method that checks both global and per-experience permission
-- (BOOL)hasGrantedPermission:(NSString *)permissionType
+- (BOOL)hasGrantedPermissionUsingRequesterClass:(Class)requesterClass
 {
-  NSDictionary *permissions = [self getPermissionsForResource:permissionType];
-
+  NSDictionary *permissions = [self getPermissionUsingRequesterClass:requesterClass];
   if (!permissions) {
-    UMLogWarn(@"Permission with type '%@' not found.", permissionType);
+    UMLogWarn(@"Permission requester '%@' not found.", NSStringFromClass(requesterClass));
     return false;
   }
   
   return [permissions[@"status"] isEqualToString:@"granted"];
 }
 
-- (void)askForPermission:(NSString *)permissionType
-              withResult:(void (^)(NSDictionary *))onResult
-            withRejecter:(UMPromiseRejectBlock)reject
+// From UMPermissionsInterface - use to check permission from code
+// shorthand method that checks both global and per-experience permission
+- (BOOL)hasGrantedPermission:(NSString *)permissionType
 {
-  return [self askForPermissions:@[permissionType]
-                     withResults:^(NSArray<NSDictionary *> *results) {
-                       onResult(results[0]);
-                     }
-                    withRejecter:reject];
+  NSDictionary *permissions = [self getPermissionsForResource:permissionType];
+  if (!permissions) {
+    UMLogWarn(@"Permission with type '%@' not found.", permissionType);
+    return false;
+  }
+  
+  return [permissions[@"status"] isEqualToString:@"granted"];
 }
 
-- (void)askForPermissions:(NSArray<NSString *> *)permissionsTypes
-              withResults:(void (^)(NSArray<NSDictionary *> *))onResults
-             withRejecter:(UMPromiseRejectBlock)reject
+- (void)askForPermissionUsingRequesterClass:(Class)requesterClass
```

Comment:
This method will handle requesting permission once requesters would be moved to separate modules, right?